src/Response/Handler/ResponseHandler.php

```diff
@@ -18,21 +19,22 @@ class ResponseHandler implements ResponseHandlerInterface
     /**
      * @param ResponseInterface $response
      *
-     * @return array
+     * @return ResponseData
      *
      * @throws NotFoundResponseException
      * @throws BadRequestResponseException
      * @throws UnexpectedResponseException
      * @throws JsonException
      */
-    public function handle(ResponseInterface $response): array
+    public function handle(ResponseInterface $response): ResponseData
     {
         $statusCode = $response->getStatusCode();
         $body       = $response->getBody();
+        $headers    = $response->getHeaders();
 
         if ($statusCode >= 200 && $statusCode < 300) {
             if ($body === null || (int)$body->getSize() === 0) {
-                return [];
+                return new ResponseData([], $headers);
```

Comment:
What do you think about just Response as the name of the class?

- [ ] convider naming alternative

Reply:
I agree, it's done